src/test/java/org/junit/tests/assertion/AssertionTest.java

```diff
@@ -367,9 +375,11 @@ public void arraysWithNullElementEqual() {
     public void stringsDifferWithUserMessage() {
         try {
             assertEquals("not equal", "one", "two");
-        } catch (Throwable exception) {
+        } catch (AssertionError exception) {
```

Comment:
You can change this to catch `ComparisonFailure` instead (in which case, it's perfectly fine to call `fail()` in the `try` block)